File: J9Ch/src/J_7_Two_Pointers/Optional_11/python_cpp/_604_Window_Sum_Easy.cpp
/** 604 Window Sum */


class Solution {
public:
    /**
     * @param nums a list of integers.
     * @return the sum of the element inside the window at each moving
     */
    vector<int> winSum(vector<int> &nums, int k) {
        // write your code here
        if (nums.size() < k || k <= 0)
            return vector<int>();

        int n = nums.size();
        vector<int> sums(n - k + 1, 0);

        for (int i = 0; i < k; i++)
            sums[0] += nums[i];
        for (int i = 1; i < n - k + 1; i++) {
            sums[i] = sums[i-1] - nums[i-1] + nums[i + k-1];
        }
        return sums;
    }
};

File: J9Ch/src/J_7_2Pointers/Optional_11/python_cpp/_604_Window_Sum_Easy.py
#/** 604 Window Sum


class Solution:
    # @param nums {int[]} a list of integers
    # @param k {int} size of window
    # @return {int[]} the sum of element inside the window at each moving
    def winSum(self, nums, k):
        # Write your code here
        n = len(nums)
        if n < k or k <= 0:
            return []
        sums = [0] * (n - k + 1)
        for i in xrange(k):
            sums[0] += nums[i];

        for i in xrange(1, n - k + 1):
            sums[i] = sums[i - 1] - nums[i - 1] + nums[i + k - 1]

        return sums